ksqldb-serde/src/test/java/io/confluent/ksql/serde/json/KsqlJsonSchemaDeserializerTest.java

```diff
@@ -0,0 +1,149 @@
+          .build())
+      .field(MAPCOL, SchemaBuilder
+          .map(Schema.OPTIONAL_STRING_SCHEMA, Schema.OPTIONAL_FLOAT64_SCHEMA)
+          .optional()
+          .build())
+      .field(TIMEFIELD, ConnectKsqlSchemaTranslator.OPTIONAL_TIME_SCHEMA)
+      .field(DATEFIELD, ConnectKsqlSchemaTranslator.OPTIONAL_DATE_SCHEMA)
+      .field(TIMESTAMPFIELD, ConnectKsqlSchemaTranslator.OPTIONAL_TIMESTAMP_SCHEMA)
+      .field(BYTESFIELD, Schema.OPTIONAL_BYTES_SCHEMA)
+      .version(0)
+      .build();
+
+  private static final Struct AN_ORDER = new Struct(ORDER_SCHEMA)
+      .put(ORDERTIME, 1511897796092L)
+      .put(ORDERID, 1L)
+      .put(ITEMID, "Item_1")
+      .put(ORDERUNITS, 10.0)
+      .put(ARRAYCOL, ImmutableList.of(10.0, 20.0))
+      .put(MAPCOL, ImmutableMap.of("key1", 10.0))
+      .put(CASE_SENSITIVE_FIELD, 1L)
+      .put(TIMEFIELD, new java.sql.Time(1000))
+      .put(DATEFIELD, new java.sql.Date(864000000L))
+      .put(TIMESTAMPFIELD, new java.sql.Timestamp(1000))
+      .put(BYTESFIELD, ByteBuffer.wrap(new byte[] {123}));
+
+  private Struct expectedOrder;
+  private Serializer<Struct> serializer;
+  private Deserializer<Struct> deserializer;
+  private Converter converter;
+  private SchemaRegistryClient schemaRegistryClient;
+  private ParsedSchema schema;
+
+  @Before
+  public void before() throws Exception {
+    schema = (new JsonSchemaTranslator()).fromConnectSchema(ORDER_SCHEMA.schema());
+
+    schemaRegistryClient = mock(SchemaRegistryClient.class);
+    when(schemaRegistryClient.getSchemaBySubjectAndId(anyString(), anyInt())).thenReturn(schema);
+
+    converter = new JsonSchemaConverter(schemaRegistryClient);
+    converter.configure(ImmutableMap.of(
+        "schema.registry.url", "dummyhost",
+        JsonConverterConfig.DECIMAL_FORMAT_CONFIG, DecimalFormat.NUMERIC.name(),
+        JsonSchemaDataConfig.GENERALIZED_SUM_TYPE_SUPPORT_CONFIG, true
+    ), false);
+
+    expectedOrder = new Struct(ORDER_SCHEMA)
+        .put(ORDERTIME, 1511897796092L)
+        .put(ORDERID, 1L)
+        .put(ITEMID, "Item_1")
+        .put(ORDERUNITS, 10.0)
+        .put(ARRAYCOL, ImmutableList.of(10.0, 20.0))
+        .put(MAPCOL, ImmutableMap.of("key1", 10.0))
+        .put(CASE_SENSITIVE_FIELD, 1L)
+        .put(TIMEFIELD, new java.sql.Time(1000))
+        .put(DATEFIELD, new java.sql.Date(864000000L))
+        .put(TIMESTAMPFIELD, new java.sql.Timestamp(1000))
+        .put(BYTESFIELD, ByteBuffer.wrap(new byte[] {123}));
+
+    final KsqlJsonSerdeFactory jsonSerdeFactory =
+        new KsqlJsonSerdeFactory(new JsonSchemaProperties(ImmutableMap.of()));
+
+    final Serde<Struct> serde = jsonSerdeFactory.createSerde(
+        ORDER_SCHEMA,
+        new KsqlConfig(ImmutableMap.of()),
+        () -> schemaRegistryClient,
+        Struct.class,
+        false
+    );
+
+    serializer = serde.serializer();
+    deserializer = serde.deserializer();
+  }
+
+  @Test
+  public void shouldDeserializeJsonObjectCorrectly() {
```

Comment:
Can you add a test for `validateSchema` failure?

Reply:
The `validateSchema` test was already in `KsqlJsonSerdeFactoryTest`. I added the fflag test too.